### DISCUSSION TITLE
macOS only formula

### DISCUSSION THREAD

**Discussion by quackerex** [Homebrew/discussions#4914] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I have a formula that only works on macOS. I have set macOS and Xcode as dependencies, but the brew test bot still try to build that on linux. How do I set a formula so it only build on macOS?

**Comment by SMillerDev** [comment]
If it actually tries to build your dependency declaration is incorrect. It will always run the Linux action though

**Reply by quackerex** [reply]
This is what I have.
```rb
depends_on xcode: ["15.0.1", :build]
depends_on macos: :sonoma
uses_from_macos "swift"
```
For some reason the linux action install swift from brew in ubuntu even though it explicitly tell to use macOS Sonoma and swift from macOS.

**Reply by quackerex** [reply]
The reason for failing is that brew swift is not up to date. Still why is it trying to build in the first place?

**Reply by Bo98** [reply, marked as answer]
If you want to restrict it to macOS, you'll need to add `depends_on :macos`. This can work in addition to what you have.

It's perhaps a little confusing, but `depends_on macos: :sonoma` just says that the minimum macOS should be Sonoma, rather than it needs to be macOS. It makes a little more sense when you consider formulae that do support both macOS and Linux, but just don't work on some older macOS. Perhaps we should have named it `depends_on minimum_macos: :sonoma`, but it comes from a time when we didn't support Linux at all.

**Reply by quackerex** [reply]
`depends_on :macos` passed the linux test by showing "Warning: macOS is required for this software".

> Perhaps we should have named it depends_on minimum_macos: :sonoma, but it comes from a time when we didn't support Linux at all.

Yeah it is bit confusing as I assumed `depends_on macos: :sonoma` to mean the formula need macOS Sonoma to work.
